File: repos_estudiantes_ALVAREZ_primera_entrega/CARDON_PAZ_LUIS_ABDEL/INTRODUCCION_POO/EJERCICIO_3/Producto.py
class Producto:
    def __init__(self, nombre, precio, stock):
        self.nombre = nombre
        self.precio = precio
        self.stock = stock

    def vender(self, cantidad):
        if cantidad <= 0:
            print("No se vendio ningun producto")
            return
        elif 0 < cantidad <= self.stock:
            self.stock -= cantidad
            print(f"Se vendio: {cantidad} unidades del producto: {self.nombre}")
        else:
            print(f"No hay stock suficiente del producto: {self.nombre} para la venta")
            return

    def reabastecer(self, cantidad):
        if cantidad <= 0:
            print("No se reabastecio ningun producto")
            return
        else:
            self.stock += cantidad
            print(f"Se reabastecio: {cantidad} unidades del producto: {self.nombre}")


producto1 = Producto("Leche", 4.5, 20)
producto2 = Producto("Pan", 0.5, 15)

print("Venta de productos")
producto1.vender(5)
producto2.vender(10)
producto1.vender(20)
producto2.vender(-3)

print("\nReabastecimiento de productos")
producto1.reabastecer(10)
producto2.reabastecer(5)
producto1.reabastecer(-4)

File: repos_estudiantes_ALVAREZ_primera_entrega/CARDON_PAZ_LUIS_ABDEL/INTRODUCCION_POO/EJERCICIO_3/Producto.java
package INTRODUCCION_POO.EJERCICIO_3;

public class Producto {
    private String nombre;
    private double precio;
    private int stock;

    public Producto(String nombre, double precio, int stock) {
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
    }
    public void vender(int cantidad){
        if(cantidad<=0){
            System.out.println("No se vendio ningun producto");
            return;
        }else if (cantidad > 0 && cantidad <= stock){
            this.stock = this.stock - cantidad;
            System.out.println("Se vendio: " + cantidad + " unidades del producto: " + this.nombre);
        }else{
            System.out.println("No hay stock suficiente del producto: " + this.nombre + " para la venta");
            return;
        }
    }

    public void reabastecer(int cantidad){
        if(cantidad <= 0){
            System.out.println("No se reabastecio ningun producto");
            return;
        }else{
            this.stock = this.stock + cantidad;
            System.out.println("Se reabastecio: " + cantidad + " unidades del producto: " + this.nombre);
        }
    }
    
}
